src/main/java/ch/njol/skript/effects/EffParse.java

```diff
@@ -0,0 +1,80 @@
+package ch.njol.skript.effects;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.classes.Changer.ChangerUtils;
+import ch.njol.skript.classes.ClassInfo;
+import ch.njol.skript.classes.Parser;
+import ch.njol.skript.lang.*;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.registrations.Classes;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+public class EffParse extends Effect {
+
+	static {
+		Skript.registerEffect(EffParse.class, "parse %~strings% as %*classinfo%");
+	}
+
+	private Expression<String> toParse;
+	private ClassInfo<?> classInfo;
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] expressions, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		toParse = (Expression<String>) expressions[0];
+		classInfo = ((Literal<ClassInfo<?>>) expressions[1]).getSingle();
+
+		if (!toParse.getAnd()) {
+			Skript.error("Can't use 'or' in a parse effect");
```

Comment:
```suggestion
			Skript.error("Can't use 'or' lists in a parse effect");
```
You could also have this be a warning and just use getAll regardless, up to you